api/src/indigo_loaders.cpp

```diff
@@ -91,6 +157,45 @@ int IndigoRdfData::getIndex()
     return _index;
 }
 
+IndigoJSONMolecule::IndigoJSONMolecule( rapidjson::Value& node, rapidjson::Value& rgroups, int index )
+: IndigoObject( JSON_MOLECULE ), _node( node ), _rgroups( rgroups ), _loaded( false )
+{
+    
+}
+
+Molecule& IndigoJSONMolecule::getMolecule()
+{
+    if( !_loaded )
+    {
+        MoleculeJsonLoader loader( _node, _rgroups );
+        loader.loadMolecule( _mol );
+        _loaded = true;
+    }
+    return _mol;
+}
+
+BaseMolecule& IndigoJSONMolecule::getBaseMolecule()
+{
+    return getMolecule();
+}
+
+IndigoObject* IndigoJSONMolecule::clone()
+{
+    return IndigoMolecule::cloneFrom(*this);
+}
+
+
+const char* IndigoJSONMolecule::getName()
+{
+    if (getMolecule().name.ptr() == 0)
+        return "";
+    return getMolecule().name.ptr();
+}
+
+IndigoJSONMolecule::~IndigoJSONMolecule()
+{
+}
+

```

Comment:
It's hard to identify among a large pull-request that these strings are just moved. 
Maybe keep them in-place?